src/Instruments/financial_instruments.jl

```diff
@@ -227,12 +225,12 @@ AmericanCallOption(;kwargs...)
 ```
 """
 AmericanCallOption(
-    widget::Widget,
-    strike_price::Real = widget.prices[end];
+    widget,
+    strike_price = widget.prices[end];
     maturity = 1,
     risk_free_rate = 0.02,
     label = "",
-    values_library = Dict{String,Dict{String,AbstractFloat}}(),
+    values_library = Dict{String,Dict{String,Float64}}(),
```

Comment:
But this one is not : you remove the type-unstable AbstractFloat from the definition of the struct, but you replaced it with a very-restrictive concrete FLoat64. See my comment on PR #72 for the right way to : 
- Keep concretely typed objects, so that the compiler can go through your code (huge speedup)
- Allow composability by using types parameters